3rdparty/mlperftiny/api/internally_implemented.cpp

```diff
@@ -0,0 +1,325 @@
+
+  if (strncmp(tok, EE_CMD_NAME, EE_CMD_SIZE) == 0) {
+    th_printf(EE_MSG_NAME, EE_DEVICE_NAME, TH_VENDOR_NAME_STRING);
+  } else if (strncmp(tok, EE_CMD_TIMESTAMP, EE_CMD_SIZE) == 0) {
+    th_timestamp();
+  } else if (ee_profile_parse(tok) == EE_ARG_CLAIMED) {
+  } else {
+    th_printf(EE_ERR_CMD, tok);
+  }
+
+  th_printf(EE_MSG_READY);
+}
+
+/**
+ * Perform the basic setup.
+ */
+void ee_benchmark_initialize(void) {
+  th_serialport_initialize();
+  th_timestamp_initialize();
+  th_final_initialize();
+  th_printf(EE_MSG_INIT_DONE);
+  // Enable the command parser here (the callback is connected)
+  g_state_parser_enabled = true;
+  // At this point, the serial monitor should be up and running,
+  th_printf(EE_MSG_READY);
+}
+
+arg_claimed_t ee_profile_parse(char *command) {
+  char *p_next; /* strtok already primed from ee_main.c */
+
+  if (strncmp(command, "profile", EE_CMD_SIZE) == 0) {
+    th_printf("m-profile-[%s]\r\n", EE_FW_VERSION);
+    th_printf("m-model-[%s]\r\n", TH_MODEL_VERSION);
+  } else if (strncmp(command, "help", EE_CMD_SIZE) == 0) {
+    th_printf("%s\r\n", EE_FW_VERSION);
+    th_printf("\r\n");
+    /* These are the three common functions for all IoTConnect f/w. */
+    th_printf("help         : Print this information\r\n");
+    th_printf("name         : Print the name of the device\r\n");
+    th_printf("timestsamp   : Generate a timetsamp\r\n");
+    /* These are profile-specific commands. */
+    th_printf("db SUBCMD    : Manipulate a generic byte buffer\r\n");
+    th_printf("  load N     : Allocate N bytes and set load counter\r\n");
+    th_printf("  db HH[HH]* : Load 8-bit hex byte(s) until N bytes\r\n");
+    th_printf("  print [N=16] [offset=0]\r\n");
+    th_printf("             : Print N bytes at offset as hex\r\n");
+    th_printf(
+        "infer N [W=0]: Load input, execute N inferences after W warmup "
+        "loops\r\n");
+    th_printf("results      : Return the result fp32 vector\r\n");
+  } else if (ee_buffer_parse(command) == EE_ARG_CLAIMED) {
+  } else if (strncmp(command, "infer", EE_CMD_SIZE) == 0) {
+    size_t n = 1;
+    size_t w = 10;
+    int i;
+
+    /* Check for inference iterations */
+    p_next = strtok(NULL, EE_CMD_DELIMITER);
+    if (p_next) {
+      i = atoi(p_next);
+      if (i <= 0) {
+        th_printf("e-[Inference iterations must be >0]\r\n");
+        return EE_ARG_CLAIMED;
+      }
+      n = (size_t)i;
+      /* Check for warmup iterations */
+      p_next = strtok(NULL, EE_CMD_DELIMITER);
+      if (p_next) {
+        i = atoi(p_next);
+        if (i < 0) {
+          th_printf("e-[Inference warmup must be >=0]\r\n");
+          return EE_ARG_CLAIMED;
+        }
+        w = (size_t)i;
+      }
+    }
+
+    ee_infer(n, w);
+  } else if (strncmp(command, "results", EE_CMD_SIZE) == 0) {
+    th_results();
+  } else {
+    return EE_ARG_UNCLAIMED;
+  }
+  return EE_ARG_CLAIMED;
+}
+
+/**
+ * Inference without feature engineering. The inpput tensor is expected to
```

Comment:
nit: 
```suggestion
 * Inference without feature engineering. The input tensor is expected to
```

Reply:
I guess that needs to be fixed upstream.

Reply:
yeah, I haven't changed the API files.